walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -27,35 +27,41 @@
   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
 
-    ArrayList<TimeRange> allUnavailableTimes = new ArrayList<>();
+    ArrayList<TimeRange> mandatoryUnavailableTimes = new ArrayList<>();
+    ArrayList<TimeRange> optionalUnavailableTimes = new ArrayList<>();
 
     for (Event event : events){
-      if (!isEventBlockingRequest(event, request)){
-        allUnavailableTimes.add(event.getWhen());
-      }
-    }
 
-    ArrayList<TimeRange> mergedUnavailableTimes = mergeTimeRanges(allUnavailableTimes);
-    ArrayList<TimeRange> availableTimes = new ArrayList<>();
-    
-    TimeRange possibleTime;
-    int prevEndTime = TimeRange.START_OF_DAY;
-    
-    for (TimeRange curTime: mergedUnavailableTimes){
-
-      possibleTime = TimeRange.fromStartEnd(prevEndTime, curTime.start(), false);
+      if (!isEventBlockingRequest(event, request, true)){
```

Comment:
Nit but here and in the rest of your code you should include a space before the curly brace, it's one of the Java style guidelines! (put this in the wrong place the first time)

Reply:
Done!